### PR TITLE
Upgrading IncludingJSON to IncludingNetworkResponse

### PR DESCRIPTION
**Pull request by LeffelMania**
I couldn't come up with an acceptable way to _compose_ the `IncludingJSON` and a new `IncludingHTTP` request struct (they're structs, so there's no inheritance, and the functionality doesn't come from a protocol I could extend, so it seemed like I would've had to create separate structs for JSON, HTTP, JSONAndHTTP, which seemed not good). So I beefed up the existing `IncludingJSON` struct to be a more encompassing `IncludeNetworkResponse` struct where I've dumped the json payload and the HTTP response properties. I had to make my own `HTTPResponse` class because I couldn't tell the existing Foundation `HTTPURLResponse` class to conform to `Decodable`.

Thoughts welcome, I know this is not what was expected from our conversation earlier this week.

### CODE REVIEW

**Comment by LeffelMania** [comment]
P.S. I've pointed my Remind branch at this branch to test out, works as expected with just the `send(...)` label change.

File: AutoGraphTests/AutoGraphTests.swift [hunk 1]
```diff
@@ -247,6 +247,7 @@ class AutoGraphTests: XCTestCase {
...
+            XCTAssertEqual(data.httpResponse?.statusCode, 200)
```
Comment (rex-remind101) [inline AutoGraphTests/AutoGraphTests.swift:250]: can you add an assert for included headers and url string too?

File: AutoGraph/ResponseHandler.swift [hunk 2]
```diff
@@ -40,16 +40,28 @@ open class ResponseHandler {
...
+                        case true:
+                            var result: [String : JSONValue] = ["json" : objectJson, "value" : objectJson]
+                            if let httpResponse = response {
+                                let http = try? JSONValue(dict: [
+                                    "urlString" : httpResponse.url?.absoluteString ?? "",
+                                    "statusCode" : httpResponse.statusCode,
+                                    "headerFields" : httpResponse.allHeaderFields
+                                ])
+                                result["httpResponse"] = http
+                            }
```
Comment (rex-remind101) [inline AutoGraph/ResponseHandler.swift:60]: could invert this and result to keep things immutable.
```swift
let httpResponseJson = response.map {
    try? JSONValue(dict: [
        "urlString" : $0.url?.absoluteString ?? "",
        "statusCode" : $0.statusCode,
        "headerFields" : $0.allHeaderFields
    ])
} ?? .null

let result: [String : JSONValue] = [
    "json" : objectJson,
    "value" : objectJson,
    "httpResponse" : httpResponseJson
]
```
This also clues the ultimate receiver of this data into the fact that `httpResponse` is null, which frankly, should always be there in our setup anyway and shouldn't happen.

File: AutoGraph/Utilities.swift [hunk 3]
```diff
@@ -3,7 +3,7 @@ import Foundation
 import JSONValueRX
 
 public typealias AutoGraphResult<Value> = Swift.Result<Value, Error>
-public typealias ResultIncludingJSON<Value: Decodable> = AutoGraphResult<DataIncludingJSON<Value>>
+public typealias ResultIncludingJSON<Value: Decodable> = AutoGraphResult<DataIncludingNetworkResponse<Value>>
```
Comment (rex-remind101) [inline AutoGraph/Utilities.swift:6]: `ResultIncludingJSON` -> `ResultIncludingNetworkResponse`?

File: AutoGraph/ResponseHandler.swift (hunk 2 above)
Comment (LeffelMania) [reply AutoGraph/ResponseHandler.swift]: Nice.

**Pull request merged**